doc/reference/reference_lua/config/utils_schema.rst

```diff
@@ -0,0 +1,1163 @@
+            -   Allowed node values
+
+        *   -   :ref:`schema_node_object.apply_default_if <config-schema_node_object-apply_default_if>`
+            -   Condition to apply defaults
+
+        *   -   :ref:`schema_node_object.computed <config-schema_node_object-computed>`
+            -   Computed annotations
+
+        *   -   :ref:`schema_node_object.default <config-schema_node_object-default>`
+            -   Default value
+
+        *   -   :ref:`schema_node_object.fields <config-schema_node_object-fields>`
+            -   Record node fields
+
+        *   -   :ref:`schema_node_object.items <config-schema_node_object-items>`
+            -   Array node items
+
+        *   -   :ref:`schema_node_object.type <config-schema_node_object-type>`
+            -   Scalar node type
+
+        *   -   :ref:`schema_node_object.validate <config-schema_node_object-validate>`
+            -   Validation function
+
+
+..  _config-utils-schema_functions:
+
+Functions
+~~~~~~~~~
+
+..  module:: config.utils.schema
+
+..  _config-utils-schema-array:
+
+..  function:: schema.array(opts)
+
+    Create an array node of a configuration schema.
+
+    :param table opts: a table in the following format:
+
+                       ..  code-block:: text
+
+                           { items = <schema_node_object>, <..annotations..> }
+
+                       See also: :ref:`schema_node_object <config-utils-schema_node_object>`, :ref:`schema_node_annotation <config-utils-schema_node_annotation>`.
+
+    :return: the created schema node
+    :rtype: table
+
+    **See also:** :ref:`config_utils_schema_nodes_array`
+
+..  _config-utils-schema-enum:
+
+..  function:: schema.enum(allowed_values, annotations)
+
+    A shortcut for creating a string scalar node with a limited set of allowed values.
+
+    :param table allowed_values: a list of enum members -- values allowed for the node
+    :param table annotations: annotations (see :ref:`schema_node_annotation <config-utils-schema_node_annotation>`)
+
+    :return: the created schema node
+    :rtype: table
+
+    **See also:** :ref:`config_utils_schema_nodes_scalar`
+
+..  _config-utils-schema-fromenv:
+
+..  function:: schema.fromenv(env_var_name, raw_value, schema_node)
+
+    Parse an environment variable as a value of the given schema node.
+    The ``env_var_name`` parameter is used only for error messages.
+    The value (``raw_value``) should be received using ``os.getenv()`` or ``os.environ()``.
+
+    How the raw value is parsed depends on the ``schema_node`` type:
+
+    -   Scalar:
+
+        -   ``string``: return the value as is
+        -   ``number`` or ``integer``: parse the value as a number or an integer
+        -   ``string, number``: attempt to parse as a number; in case of a failure
+            return the value as is
+        -   ``boolean``: accept ``true`` or ``1`` for ``true``, ``false`` or ``0`` for ``false``
```

Comment:
These `true` and `false` source values are taken case-insensitively. I think it is a part of the API and should be mentioned.

Reply:
Added